pc/manager.cpp

```diff
@@ -535,30 +586,23 @@ void manager::poll( bool do_wait )
   if ( has_status( PC_PYTH_RPC_CONNECTED ) &&
        !hconn_.get_is_err() &&
        ( !wconn_ || !wconn_->get_is_err() ) ) {
-    // request product quotes from pythd's clients while connected
-    poll_schedule();
-
     send_pending_ups();
   } else {
     reconnect_rpc();
   }
+
+  // Call the secondary manager's poll loop if necessary
+  if ( has_secondary() ) {
+    secondary_->poll();
+  }
 }
 
 void manager::poll_schedule()
 {
-  while ( is_pub_ && kidx_ < kvec_.size() ) {
-    price_sched *kptr = kvec_[kidx_];
-    int64_t pub_ts = pub_ts_ + static_cast< int64_t >(
-      ( static_cast< uint64_t >( pub_int_ ) * kptr->get_hash() )
-        / price_sched::fraction
-    );
-    if ( curr_ts_ > pub_ts ) {
+  for ( uint32_t i = 0; i < kvec_.size(); i++ ) {
+    price_sched *kptr = kvec_[i];
+    if ( ( curr_ts_ - kptr->get_last_scheduled_time() ) > pub_int_ ) {
```

Comment:
the old logic would space out the price scheduling requests in time. I think the goal of this logic was to space out the solana transactions to increase the probability of landing them. I'm pretty sure this new logic will request all of the prices in bursts. This may cause some issues with tx hit rate, though hard to say for sure. I think you can either demonstrate that it doesn't cause a problem, or make this logic space out the txs again.